chore: Update to .NET 10 Preview 7

jonpryor opened the pull request:
Context: https://github.com/unoplatform/uno/pull/19934
Context: https://github.com/unoplatform/uno/pull/21140
Context: https://github.com/dotnet/sdk/issues/48949

unoplatform/uno#19934 was the first attempt to build unoplatform/uno with .NET 10.  It *built* (mostly), but it had *lots* of unit test failures (over 3700) that were not understood at the time.

unoplatform/uno#21140 was an attempt to cleanup & simplify #19934. It partially succeeded, but required additional reconsideration for a cleaner and more understandable merge.

(Later, PR #21183 helped explain many of the unit test failures that PR #19934 observed: Xamarin.UITest doesn't work reliably under .NET 9! See microsoft/appcenter#2646 and microsoft/appcenter#327!)

Some of the contents of #19934 and #21140 have been split out into separately merged PRs such as #21230 (CS0246 fixes) and #21199 (remove `[Export]` in some scenarios).

Squash existing "known good" and related changes:

  * doc updates

  * Bumping target framework versions,

  * Use uno.check 1.32.0-dev.45, which knows about .NET 10 Preview 7.

  * `$(TreatWarningsAsErrors)`=true + NuGet warnings.  For example:

        error NU1510: Warning As Error:
        PackageReference System.Private.Uri will not be pruned.
        Consider removing this package from your dependencies, as it is likely unnecessary.

  * `$(TreatWarningsAsErrors)`=true + CS8604 warnings due to nullability changes, particularly in iOS.

  * `$(TreatWarningsAsErrors)`=true + IL2\* warnings, some of which @jonpryor was able to fix in a reasonable timeframe, and some of which was thrown into `$(NoWarn)` to deal with later.

  * Remove hardcoded `$(TargetFramework)`=net8.0 from `Uno.UI.RemoteControl.TestProcessor.csproj`, so that it builds, and remove other uses of `net8.0` from everywhere other than `src/SolutionTemplate`.

  * Provide `%(PackageReference.Version)` for packages that were producing NU1015 errors, specifically `Uno.Fonts.OpenSans`.

  * Remove `@(PackageReference)` values which produced NU1510 warnings such as:

        warning NU1510: PackageReference Microsoft.Win32.Registry will not be pruned. Consider removing this package from your dependencies, as it is likely unnecessary.

  * Improve `dotnet-install.yml` use, as some CI environments *don't* have a `unoplatform/uno` checkout, and thus *cannot* copy files such as `build/ci/net10/global.json`.

  * Improve `local-android-uitest-run.sh` for local macOS use: Export and use `$ANDROID_AVD_HOME`, so that we have a "known" location for whare Emulator images are created.  (Not sure what's wrong with my environment, but they are not reliably created or loaded from `$HOME/.android/avd`.)

    Also update so that it doesn't require running the script from the `build/test-scripts` directory.  You can now run it from topdir:

        UITEST_IS_LOCAL=true \
          build/test-scripts/local-android-uitest-run.sh

  * Provision OpenJDK-17, as .NET 9 now requires it.

  * Use Xcode 16.4, as .NET 10 iOS now requires it.

TODO, for future commits to this PR:

  * Xamarin.AndroidX package version bumps from #21205. We felt that these shouldn't be separately merged, as some of version bumps are quite significant and should instead only be used with .NET 10.

  * A new understanding of `src/SolutionTemplate`: the intent, as @jonpryor currently understands it, is that tests within `src/SolutionTemplate` are "frozen in time"; `$(TargetFrameworks)` should *not* be updated, etc.

    Which means that when we drop support for an older version of .NET, we should in turn *remove the old tests*.

**GitHub Issue:** closes #

<!-- Link to relevant GitHub issue if applicable. All PRs should be associated with an issue (GitHub issue or internal), unless the change is documentation related. -->

## PR Type:

<!--
Copy the labels that apply to this PR and paste them above:

- 🐞 Bugfix
- ✨ Feature
- 🎨 Code style update (formatting)
- 🔄 Refactoring (no functional changes, no api changes)
- 🏗️ Build or CI related changes
- 📚 Documentation content changes
- 🤖 Project automation
- 💬 Other... (Please describe)

-->


## What is the current behavior? 🤔

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->


## What is the new behavior? 🚀

<!-- Please describe the new behavior after your modifications. -->

## PR Checklist ✅

Please check if your PR fulfills the following requirements:

- [ ] 📝 Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.
- [ ] 🧪 Added [Runtime tests, UI tests, or a manual test sample](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] 📚 Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] 🖼️ Validated PR `Screenshots Compare Test Run` results.
- [ ] ❗ Contains **NO** breaking changes

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below.
     Please note that breaking changes are likely to be rejected -->

## Other information ℹ️

<!-- Please provide any additional information if necessary -->

jonpryor committed:
chore: Update to .NET 10 Preview 7

Context: https://github.com/unoplatform/uno/pull/19934
Context: https://github.com/unoplatform/uno/pull/21140
Context: https://github.com/dotnet/sdk/issues/48949

unoplatform/uno#19934 was the first attempt to build unoplatform/uno
with .NET 10.  It *built* (mostly), but it had *lots* of unit test
failures (over 3700) that were not understood at the time.

unoplatform/uno#21140 was an attempt to cleanup & simplify #19934.
It partially succeeded, but required additional reconsideration for
a cleaner and more understandable merge.

(Later, PR #21183 helped explain many of the unit test failures that
PR #19934 observed: Xamarin.UITest doesn't work reliably under
.NET 9! See microsoft/appcenter#2646 and microsoft/appcenter#327!)

Some of the contents of #19934 and #21140 have been split out into
separately merged PRs such as #21230 (CS0246 fixes)
and #21199 (remove `[Export]` in some scenarios).

Squash existing "known good" and related changes:

  * doc updates

  * Bumping target framework versions,

  * Use uno.check 1.32.0-dev.45, which knows about .NET 10 Preview 7.

  * `$(TreatWarningsAsErrors)`=true +
    NuGet warnings.  For example:

        error NU1510: Warning As Error:
        PackageReference System.Private.Uri will not be pruned.
        Consider removing this package from your dependencies, as it is likely unnecessary.

  * `$(TreatWarningsAsErrors)`=true + CS8604 or CS0809 warnings due
    to nullability changes, particularly in iOS.

  * `$(TreatWarningsAsErrors)`=true + IL2\* warnings, some of which
    @jonpryor was able to fix in a reasonable timeframe, and some of
    which was thrown into `$(NoWarn)` to deal with later.

  * Remove hardcoded `$(TargetFramework)`=net8.0 from
    `Uno.UI.RemoteControl.TestProcessor.csproj`, so that it builds,
    and remove other uses of `net8.0` from everywhere other than
    `src/SolutionTemplate`.

  * Provide `%(PackageReference.Version)` for packages that were
    producing NU1015 errors, specifically `Uno.Fonts.OpenSans`.

  * Remove `@(PackageReference)` values which produced NU1510
    warnings such as:

        warning NU1510: PackageReference Microsoft.Win32.Registry will not be pruned. Consider removing this package from your dependencies, as it is likely unnecessary.

  * Improve `dotnet-install.yml` use, as some CI environments *don't*
    have a `unoplatform/uno` checkout, and thus *cannot* copy files
    such as `build/ci/net10/global.json`.

  * Improve `local-android-uitest-run.sh` for local macOS use:
    Export and use `$ANDROID_AVD_HOME`, so that we have a "known"
    location for whare Emulator images are created.  (Not sure what's
    wrong with my environment, but they are not reliably created or
    loaded from `$HOME/.android/avd`.)

    Also update so that it doesn't require running the script from the
    `build/test-scripts` directory.  You can now run it from topdir:

        UITEST_IS_LOCAL=true \
          build/test-scripts/local-android-uitest-run.sh

  * Provision OpenJDK-17, as .NET 9 requires it.

  * Use Xcode 16.4, as .NET 10 iOS now requires it.

  * Add new `$(NetPreviousNetCoreApple)` MSBuild property to
    `Directory.Build.props`, which contains target frameworks for
    $(NetPrevios) Apple platforms.

    Update `Uno.UI.Runtime.Skia.AppleUIKit.csproj` to use
    `$(NetPreviousNetCoreApple)`, and update `skia-ios-uitest-build.sh`
    to provide a "complete" target framework version.

    This is so that target frameworks are consistent:
    `net9.0-ios` *floats*; circa a year ago, it aliased
    `net9.0-ios18.0`, and *today* it aliases `net9.0-ios18.5`.
    "Even better", this doesn't "flow" to the C# compiler:
    `net9.0-ios` sets `IOS18_0_OR_GREATER` *but not*
    `IOS18_5_OR_GREATER` *even when* the 18.5 assemblies are used!
    This can make it annoyingly difficult to get `[Obsolete]`s
    consistent across separate project builds.

TODO, for future commits to this PR:

  * Xamarin.AndroidX package version bumps from #21205.
    We felt that these shouldn't be separately merged, as some of
    version bumps are quite significant and should instead only be
    used with .NET 10.

  * A new understanding of `src/SolutionTemplate`: the intent, as
    @jonpryor currently understands it, is that tests within
    `src/SolutionTemplate` are "frozen in time"; `$(TargetFrameworks)`
    should *not* be updated, etc.

    Which means that when we drop support for an older version of .NET,
    we should in turn *remove the old tests*.

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21256/wasm-skia-net9/index.html

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21256/wasm-skia-net9/index.html

jeromelaban commented on src/SamplesApp/SamplesApp.UITests/SamplesApp.UITests.csproj:
```diff
@@ -27,9 +27,7 @@
 		<PackageReference Include="SkiaSharp.NativeAssets.Linux" />
-		<PackageReference Include="System.Text.Json" Version="8.0.5" />
 		<PackageReference Include="Microsoft.Testing.Extensions.HangDump" />
-		<PackageReference Include="Microsoft.Testing.Extensions.VSTestBridge" />
```
Many thanks!

jonpryor replied:
@Youssef1313? what's the fix, and/or a timeline for a fix? We can't run the **Tests - WebAssembly Native > UI Tests > Running UI Tests** job because of the [`MissingMethodException`](https://github.com/unoplatform/uno/pull/21256#issuecomment-3188316808).

jonpryor commented on src/SamplesApp/SamplesApp.UITests/SamplesApp.UITests.csproj:
```diff
@@ -27,9 +27,8 @@
-		<PackageReference Include="System.Text.Json" Version="8.0.5" />
 		<PackageReference Include="Microsoft.Testing.Extensions.HangDump" />
-		<PackageReference Include="Microsoft.Testing.Extensions.VSTestBridge" />
+		<PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.14.1" />
```
This change has been reverted. (It *also* didn't fix the `MissingMethodException`…)

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21256/wasm-skia-net9/index.html

jonpryor committed:
chore(deps): Update Uno.Wasm.Bootstrap to 10.0.0-dev.24

The **Tests - WebAssembly Skia > Run Runtime Tests \* > Run WebAssembly Skia Runtime Tests**
job fails with:

	Error: Failed to load config file blazor.boot.json SyntaxError: Unexpected token '<', "<!DOCTYPE "... is not valid JSON SyntaxError: Unexpected token '<', "<!DOCTYPE "... is not valid JSON
	    at xe (http://localhost:8000/_framework/dotnet.js:4:21471)
	    at async http://localhost:8000/_framework/dotnet.js:4:31096
	    at async Bootstrapper.bootstrap (http://localhost:8000/package_cad083c022126debd2f8488f4d32c39d75cb8f7b/uno-bootstrap.js:159:47)", source: http://localhost:8000/_framework/dotnet.js (4)

This is because we need to use a newer version of the
Uno.Wasm.Bootstrap NuGet package.

Update `Directory.Build.targets` to use Uno.Wasm.Bootstrap 10.0.0-dev.24
instead of 9.0.16.

jonpryor committed:
chore: bump Xamarin.AndroidX Package Versions

Context: https://github.com/unoplatform/uno/pull/21205
Context: https://github.com/unoplatform/uno/pull/21140
Context: https://github.com/unoplatform/uno-private/issues/1422
Context: https://github.com/dotnet/java-interop/commit/356485ee41fb61570e84c11ba120d6f8cf81ae09

While exploring the use of NativeAOT on Android for .NET 10, a sticking
point is that running within an Android+NativeAOT environment requires
the use of a new "style" of bindings; see dotnet/java-interop@356485ee.

The new style requires .NET 10.

However, the various Xamarin.AndroidX packages have been updated to
provide bindings for *both* .NET 8 *and* .NET 10 previews, which allows
apps targeting .NET 10 to begin exploring NativeAOT use.

Update the required Xamarin.AndroidX NuGet package versions to versions
which provide both .NET 8 and .NET 10 binaries, but *only* for
.NET 10 target frameworks.  .NET 9 targets continue. using the previous
NuGet package versions.

Additionally, the Xamarin.AndroidX.Browser NuGet package:
[Xamarin.AndroidX.Browser/1.0.0][0] has a dependency on
Xamarin.AndroidX.Legacy.Support.Core.UI, which eventully pulls in
Xamarin.AndroidX.DocumentFile. [Xamarin.AndroidX.Browser/1.8.0.10][1]
*does not*.  The Xamarin.AndroidX.Legacy.Support.Core.UI dependency
was removed in Xamarin.AndroidX.Browser/1.2.0.

Fix a CS0246 error in `StorageFolder.saf.Android.cs` by explicitly
adding a reference to Xamarin.AndroidX.DocumentFile.

Finally, fix some nullability and `[Obsolete]` errors caused by using
a newer Xamarin.AndroidX.DocumentFile package.

[0]: https://www.nuget.org/packages/Xamarin.AndroidX.Browser/1.0.0#dependencies-body-tab
[1]: https://www.nuget.org/packages/Xamarin.AndroidX.Browser/1.8.0.10#dependencies-body-tab

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21256/wasm-skia-net9/index.html

unodevops commented:
⚠️⚠️ The build [173702](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=173702) has failed on Uno.UI - CI.

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21256/wasm-skia-net9/index.html

jonpryor committed:
chore: Remove net8.0 and earlier SolutionTemplate tests

Context: https://github.com/unoplatform/uno/pull/21256
Context: https://github.com/dotnet/android/issues/10423

PR #21256 noted:

> * A new understanding of `src/SolutionTemplate`: the intent, as
>   @jonpryor currently understands it, is that tests within
>   `src/SolutionTemplate` are "frozen in time"; `$(TargetFrameworks)`
>   should *not* be updated, etc.
>
>   Which means that when we drop support for an older version of .NET,
>   we should in turn *remove the old tests*.

.NET mobile workloads (iOS, Android, MAUI, etc.) follow the
[Modern Lifecycle Policy][0], as outlined in
[.NET Multi-platform App UI (.NET MAUI) Support Policy][1].
Support for .NET mobile workloads ends six months after a major version
release.  Support for the .NET 8 mobile workloads ended six months
after the release of .NET 9, on 2025-May-14.  Similarly, support for
the .NET 9 mobile workloads will end 6 months after the release of
.NET 10, on 2026-May-12.

Which is a long-winded way of saying that the .NET 8 mobile workloads
are no longer supported by Microsoft.

This in turn means that some projects fail to build on
**Tests - Templates > \* > Run Project Template Tests**, e.g.:

	C:\a\1\s\.dotnet\sdk\10.0.100-preview.7.25380.108\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.EolTargetFrameworks.targets(38,5):
	error NETSDK1202: The workload 'net8.0-maccatalyst' is out of support and will not receive security updates in the future.
	Please refer to https://aka.ms/maui-support-policy for more information about the support policy.
	[C:\a\1\s\src\SolutionTemplate\5.1\uno51blank\uno51blank.Mobile\uno51blank.Mobile.csproj::TargetFramework=net8.0-maccatalyst]

This is nominally "ignorable" by setting `$(CheckEolWorkloads)`=false
and `$(CheckEolTargetFramework)`=false.

Unfortunately, that's not the only error.  There's also:

	error NU1202: Package Uno.WinUI 6.3.0-dev.71.gafa7bb848f is not compatible with net8.0-android21.0 (.NETCoreApp,Version=v8.0). Package Uno.WinUI 6.3.0-dev.71.gafa7bb848f supports:
	error NU1202:   - net10.0 (.NETCoreApp,Version=v10.0)
	error NU1202:   - net10.0-android36.0 (.NETCoreApp,Version=v10.0)
	error NU1202:   - net10.0-ios18.5 (.NETCoreApp,Version=v10.0)
	error NU1202:   - net10.0-maccatalyst18.5 (.NETCoreApp,Version=v10.0)
	error NU1202:   - net10.0-tvos18.5 (.NETCoreApp,Version=v10.0)
	error NU1202:   - net10.0-windows10.0.19041 (.NETCoreApp,Version=v10.0)
	error NU1202:   - net9.0 (.NETCoreApp,Version=v9.0)
	error NU1202:   - net9.0-android30.0 (.NETCoreApp,Version=v9.0)
	error NU1202:   - net9.0-ios18.0 (.NETCoreApp,Version=v9.0)
	error NU1202:   - net9.0-maccatalyst18.0 (.NETCoreApp,Version=v9.0)
	error NU1202:   - net9.0-tvos18.0 (.NETCoreApp,Version=v9.0)
	error NU1202:   - net9.0-windows10.0.19041 (.NETCoreApp,Version=v9.0)

This is because `src/Uno.UI` is only built against `$(NetPrevious)`
and `$(NetCurrent)` -- now net9.0 and net10.0 -- which means net8.0
now "falls off" the list of supported versions.

As we are dropping support for .NET 8 with the adoption of .NET 10,
remove the `src/SolutionTemplate` tests which target .NET 8.

Additionally, provision Android API-35 on Linux and macOS, not API-34,
as .NET 9 requires API-35.

Finally, update `run-netcore-mobile-template-tests.ps1` to:

  * Add a `Write-Host` for the actually executed command.
    Relying on `Set-PSDebug -Trace 1` is less useful because the log
	messages don't expand variables:

	    DEBUG:   97+          >>>> dotnet build $debug "$projectPath" $projectParameters -bl:binlogs/$projectPath/$i/debug.binlog

    What was *actually* executed?  What does `$debug` expand to?

  * Perform a `dotnet build` *before* trying `dotnet build -r RID`.
    Works around dotnet/android#10423.

[0]: https://learn.microsoft.com/lifecycle/policies/modern
[1]: https://dotnet.microsoft.com/platform/support/policy/maui

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21256/wasm-skia-net9/index.html

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21256/wasm-skia-net9/index.html

unodevops commented:
⚠️⚠️ The build [173789](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=173789) has failed on Uno.UI - CI.

jonpryor committed:
chore: fix NU1510 warning-as-error

The **Build - Skia Runtime Tests > Build Generic Skia Head** step
is failing:

	D:\a\1\s\src\Uno.UI.Runtime.Skia.Wpf\Uno.UI.Runtime.Skia.Wpf.csproj :
	error NU1510: Warning As Error: PackageReference Microsoft.Win32.SystemEvents will not be pruned. Consider removing this package from your dependencies, as it is likely unnecessary.
	[D:\a\1\s\src\SamplesApp\SamplesApp.Skia.Generic\SamplesApp.Skia.Generic.csproj]

Do as suggested, and remove the `@(PackageReference)` for
`Microsoft.Win32.SystemEvents`.

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21256/wasm-skia-net9/index.html

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21256/wasm-skia-net9/index.html

unodevops commented:
⚠️⚠️ The build [174459](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=174459) has failed on Uno.UI - CI.

jonpryor committed:
docs: build net9.0-ios18.0 and net9.0-tvos18.0, not 17.0

Context: https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=174458&view=logs&j=204b6300-98d8-514b-21c4-843dcd1b5a33&t=6cd345c8-3621-5bcf-ae49-47c4fc86e1be

The **Uno.UI - docs > Docs Generation > Build Documenentation Website > MSBuild**
task is failing:

        [Warning] Found project reference without a matching metadata reference: C:\a\1\s\src\Uno.Foundation.Logging\Uno.Foundation.Logging.csproj
	##[error]src\Uno.UI\Uno.UI.netcoremobile.csproj(0,0): Error MSB4057: The target "Compile" does not exist in the project.
	##[error]Build\Uno.UI.Build.csproj(150,3): Error MSB3073: The command "..\src\Uno.UWPSyncGenerator\Bin\Release\Uno.UWPSyncGenerator.exe "doc"" exited with code -532462766.

Plus some NuGet warnings:

	…\Microsoft.Common.CurrentVersion.targets(2433,5): warning MSB3277: Found conflicts between different versions of "Microsoft.Extensions.Logging.Abstractions" that could not be resolved. [C:\a\1\s\src\Uno.UWPSyncGenerator\Uno.UWPSyncGenerator.csproj]
	…\Microsoft.Common.CurrentVersion.targets(2433,5): warning MSB3277: There was a conflict between "Microsoft.Extensions.Logging.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60" and "Microsoft.Extensions.Logging.Abstractions, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60". [C:\a\1\s\src\Uno.UWPSyncGenerator\Uno.UWPSyncGenerator.csproj]

Which feels generally inscrutable, except for one thing that stood out
in the log file:

	1>GenerateDoc:
	    Loading for net9.0-ios17.0: Uno.UI.netcoremobile.csproj

to which @jonpryor thought 🧐: `net9.0-ios17.0` is not a valid target
framework.  `net9.0-ios18.0` is the base!

Some `git grep net9.0-ios` later, and `Generator.cs` is the only
location within unoplatform/uno which is using net9.0-ios17; everything
else has been updated to net9.0-ios18.

Update `Generator.cs` so that it builds net9.0-ios18.0, and let's
🤞 that it works!

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21256/wasm-skia-net9/index.html

jonpryor committed:
chore: remove old `@(PackageReference)`s.

The workaround is no longer required.

jonpryor committed:
chore: revert changes to doc which mention .NET 10.

.NET 10 will continue to be preview until 2025-November, so we don't
need to "jump the gun" and update *everything* to mention .NET 10…

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-21256/docs/index.html

jonpryor commented on doc/articles/migrating-from-net8-to-net9.md:
```diff
@@ -9,7 +9,7 @@ To upgrade to .NET 9:
 - First, read [What's New in .NET 9](https://learn.microsoft.com/dotnet/core/whats-new/dotnet-9/overview)
 - Install [Visual Studio 17.12 Preview 3](https://visualstudio.microsoft.com/vs/) or later and run [uno.check](xref:UnoCheck.UsingUnoCheck) with the `--pre-major` parameter to install .NET 9.
-- Change all your target framework (TFM) references from `net8.0` to `net9.0`, and `net8.0-*` to `net9.0-*`.
+- Change all your target framework (TFM) references from `net9.0` to `net10.0`, and `net9.0-*` to `net10.0-*`.
```
All `doc` changes were reverted in 3d094d4d57151380727ffa2ed22489b58ad33740.

jeromelaban committed:
chore: Log retained types/instances

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21256/wasm-skia-net9/index.html

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-21256/docs/index.html

jeromelaban committed:
chore: Adjust GC settings for wasm skia

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21256/wasm-skia-net9/index.html

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-21256/docs/index.html

jeromelaban committed:
chore: Disable `NavigationViewItem` test on net10-wasm skia

jeromelaban committed:
chore: Disable wasm skia net10 test
